R/commonMachineLearningRegression.R

```diff
@@ -114,9 +114,9 @@
     if (any(predictorsAreFactors)) {
       predictorFactorsWithUniqueLevels <- which(sapply(predictorData[, predictorsAreFactors, drop = FALSE], FUN = function(x) length(x) == length(unique(x))))
       if (length(predictorFactorsWithUniqueLevels) == 1) {
-        jaspBase:::.quitAnalysis(gettextf("The predictor %1$s is a factor with unique levels, please remove this factor as a predictor.", names(predictorFactorsWithUniqueLevels)))
+        jaspBase:::.quitAnalysis(gettextf("The features %1$s is a factor with unique levels, please remove this factor as a feature.", names(predictorFactorsWithUniqueLevels)))
       } else if (length(predictorFactorsWithUniqueLevels) > 1) {
-        jaspBase:::.quitAnalysis(gettextf("The predictors %1$s are factors with unique levels, please remove these factors as a predictor.", paste(names(predictorFactorsWithUniqueLevels), sep = "&")))
+        jaspBase:::.quitAnalysis(gettextf("The features %1$s are factors with unique levels, please remove these factors as a feature.", paste(names(predictorFactorsWithUniqueLevels), sep = "&")))
```

Comment:
Do you mean that there is only one level in these factors?

Reply:
Would something along the line

"There is only one observation in each level of the factor ... "

be as informative? I'm quite confused about each factor level is unique, because it's not about the factor level, but too few observations per cell.